scripts/host-check

```diff
@@ -1126,31 +1140,32 @@ def check_iommu() -> CheckFuncReturn:
         return (
             CheckState.ERROR,
             f"The cmd {cmd!r} failed - unable to\n"
-            "determine the network devices on the host. IOMMU is enabled.",
-        )
-    net_iommu_devices = network_devices & set(iommu_devices)
-    if not net_iommu_devices:
-        return (
-            warning_state,
-            "IOMMU enabled for vfio-pci, but no network PCI devices found.",
-        )
-    net_iommu_devs_list = []
-    for device in sorted(net_iommu_devices):
-        net_iommu_devs_list.append(
-            net_devices_dict[device] + " (" + device + ")"
+            "determine the network devices on the host.",
         )
+
+    # If in IOMMU mode then check if the network devices are in the IOMMU group
+    if iommu_devices:
+        network_devices &= set(iommu_devices)
+        if not network_devices:
```

Comment:
Per the PR description:

If IOMMU is enabled, checks that IOMMU PCI devices are available.
If IOMMU is not enabled, checks that any PCI devices are available.

but here, if IOMMU is not enabled (which we are saying is equivalent to `iommu_devices` is empty?), this check returns WARNING?

Reply:
This return CheckState.WARNING (on L1150) is hit if both
* `iommu_devices` is not empty list - indicates IOMMU is enabled (see L1147)
* There are no network devices in the IOMMU group.

Reply:
(but subject to change, cf. comment at L1110)